R/read_data.R

```diff
@@ -0,0 +1,165 @@
+#' Read in the dataset of incident case counts
+#'
+#' Each row of the table corresponds to a single facilities' cases for a
+#' reference-date/report-date/disease tuple. We want to aggregate these counts
+#' to the level of geographic aggregate/report-date/reference-date/disease.
+#'
+#' We handle two distinct cases for geographic aggregates:
+#'
+#' 1. A single state: Subset to facilities **in that state only** and aggregate
+#' up to the state level 2. The US overall: Aggregate over all facilities
+#' without any subsetting
+#'
+#' Note that we do _not_ apply exclusions here. The exclusions are applied
+#' later, after the aggregations. That means that for the US overall, we
+#' aggregate over points that might potentially be excluded at the state level.
+#' Our recourse in this case is to exclude the US overall aggregate point.
+#'
+#' @param data_path The path to the local file. This could contain a glob and
+#'   must be in parquet format.
+#' @param disease One of "COVID-19" or "Influenza"
+#' @param state_abb A two-letter uppercase abbreviation
+#' @param report_date The desired single report date
+#' @param max_reference_date,min_reference_date The first and last reference
+#'   dates, inclusive, of the timeseries
+#'
+#' @return A dataframe with one or more rows and columns `report_date`,
+#'   `reference_date`, `state_abb`, `confirm`
+#' @export
+read_data <- function(data_path,
+                      disease = c("COVID-19", "Influenza", "test"),
+                      state_abb,
+                      report_date,
+                      max_reference_date,
+                      min_reference_date) {
+  rlang::arg_match(disease)
```

Comment:
```suggestion
  disease <- rlang::arg_match(disease)
```
Doesn't `rlang::arg_match(disease)` return the value, as opposed to modify it in place? https://rlang.r-lib.org/reference/arg_match.html. Or is this just being used to check that the value passed in isn't something bad, and working with the value is done right below?

Reply:
The latter. Unlike `match.arg()`, `rlang::arg_match()` errors for anything but a full match. Any return is invisible and it's used only as a guard against invalid options, not to fill partial matches.